### PR TITLE
web3Login: add connectButton and admin page mock

### PR DESCRIPTION
**Pull request by adiusz**

### CODE REVIEW

**Comment by adiusz** [comment]
NEXT_PUBLIC_CONTRACT_ADDRESS is: `0x6D46aC1076eEB3403FEb70e84fe7395b9998FF0E`

File: components/Button/ButtonLink.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React, { ReactNode } from "react";
+import { CustomLink } from "../CustomLink";
+import { StyledSolidButton } from "./ButtonStyles";
+
+interface Props {
```
Comment (TomaszKotowski) [inline components/Button/ButtonLink.tsx:5]: Use `type` for consistency with other components.

File: components/Button/ButtonLink.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  children?: ReactNode,
+  url?: string
+}
+
+export const ButtonLink = ({ children, url }: Props) => {
+  return (
+    <CustomLink
+      // href="https://discord.gg/uKS4vjG8"
```
Comment (TomaszKotowski) [inline components/Button/ButtonLink.tsx]: Remove comments 😉

File: components/Wallet/Connect.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import { ConnectButton } from "@rainbow-me/rainbowkit";
+
+const ConnectWallet = () => {
+  // const connectWithMetamask = useMetamask();
+  // const disconnect = useDisconnect()
+  // const address = useAddress();
+
+  // useEffect(() => {
+  //   address !== undefined && toast.success('Successfully connected with MetaMask 🦊')
+  // }, [address])
```
Comment (TomaszKotowski) [inline components/Wallet/Connect.tsx]: Is this necessary now? Add a comment if you want to do anything with this in the future.

File: pages/_app.js [hunk 4]
```diff
@@ -1,9 +1,50 @@
+import { ThirdwebProvider, ChainId } from "@thirdweb-dev/react";
+import '@rainbow-me/rainbowkit/styles.css';
+import { darkTheme, getDefaultWallets, RainbowKitProvider } from "@rainbow-me/rainbowkit";
+import { chain, configureChains, createClient, WagmiConfig, } from 'wagmi';
+import { alchemyProvider } from 'wagmi/providers/alchemy';
+import { publicProvider } from 'wagmi/providers/public';
 import "../index.css"
+import Toaster from "../components/Toaster/Toaster"
+import { ThemeProvider } from "styled-components";
+import { theme } from "../theme/theme";
```
Comment (TomaszKotowski) [inline pages/_app.js]: Please, order the imports. I recommend to order by import from `module`, internal components and last `.css` files

File: pages/admin.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+  if (!address && !isConnecting) return <h1>Please connect your wallet. 💳</h1>;
+
+  if (loading) return <h1>Checking your status... 👷</h1>;
+
+  if (!allowed && !loading) return <h1>You are not allowed here. 🚨</h1>;
+
+  if (allowed && !loading) return (
+    <>
+      <h1>hello, {shortenAddress(address)}! 🎉</h1>
```
Comment (TomaszKotowski) [inline pages/admin.js]:
```suggestion
      <h1>Hello, {shortenAddress(address)}! 🎉</h1>
```

File: hooks/useWallet.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from "react";
+import { useAccount } from "wagmi";
+import { isMember } from "../utils";
+import { toast } from "react-hot-toast";
+
+const useWallet = () => {
+  const { address, isConnecting } = useAccount();
+  const [allowed, setAllowed] = useState(false);
+  const [loading, setLoading] = useState(true);
```
Comment (TomaszKotowski) [inline hooks/useWallet.js]: Can you rename this states to `isAllowed `and `isLoading `to make it consistent with `isConnecting` from the hook `useAccount`?

**Review by 99mac99 — DISMISSED** [review]

File: components/Wallet/Connect.tsx (hunk 3 above)
Comment (adiusz) [reply components/Wallet/Connect.tsx]: i forgot about this, ofc its not necessary, gj

File: components/Button/ButtonLink.tsx (hunk 2 above)
Comment (adiusz) [reply components/Button/ButtonLink.tsx]: thx :)

File: pages/_app.js (hunk 4 above)
Comment (adiusz) [reply pages/_app.js]: u are right

File: pages/admin.js (hunk 5 above)
Comment (adiusz) [reply pages/admin.js]: ok

File: hooks/useWallet.js (hunk 6 above)
Comment (adiusz) [reply hooks/useWallet.js]: thats actually a good idea :D

File: components/Button/ButtonLink.tsx (hunk 1 above)
Comment (adiusz) [reply components/Button/ButtonLink.tsx:5]: i don't get typescript, if you want to make it more TS-friendly then you would have to do it urself 😜

**Review by 99mac99 — DISMISSED** [review]

**Pull request merged**